File: NzolaWebAPI/Dtos/Utilizador/UtilizadorDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using NzolaWebAPI.Models.Enums;

namespace NzolaWebAPI.DTOs.Utilizador
{
    public class UtilizadorDto
    {
        public int Id { get; set; }
        public String NomeCompleto { get; set; }
        public String NomeUtilizador { get; set; }
        public String Email { get; set; }
        public String Biografia { get; set; }
        public EstadoAcesso Privacidade { get; set; }
        public EstadoConta EstadoConta { get; set; }
        public string? FotoPerfil { get; set; }

        public Genero Genero { get; set; }
        public DateTime DataNascimento { get; set; }
        public int Seguidores { get; set; } // Quantos seguem ele
        public int Seguindo { get; set; } // Quantos ele segue (NOVO)
        public int Publicacoes { get; set; }
        public bool JaSegues { get; set; }
    }
}

File: NzolaSPA/src/app/dtos/utilizador/utilizadorfeed/utilizador.dto.ts
export class UtilizadorDto {
  id: number = 0;
  nomeCompleto: string = '';
  nomeUtilizador: string = '';
  email: string = '';
  biografia: string = '';
  privacidade: number = 0;
  estadoConta: number = 0;
  fotoPerfil: string | null = null;
  genero: number = 0;
  dataNascimento: string = '';
  seguidores: number = 0;
  seguindo: number = 0;
  publicacoes: number = 0;
  jaSegues: boolean = false;
  concordaComTermos: boolean = false;
}

export class EstatisticasUtilizadorDto {
  seguidores: number = 0;
  seguindo: number = 0;
  publicacoes: number = 0;
}

export class UtilizadorSimplificadoDto {
  id: number = 0;
  nomeCompleto: string = '';
  nomeUtilizador: string = '';
  fotoPerfil: string | null = null;
}
